### ISSUE TITLE
GUI generate an exception with Skript v2.4-alpha1

### ISSUE THREAD

**Issue by HevelMc** [SkQuery/SkQuery#13]
**Describe the bug**
The server console is constantly spammed when a player clicks on a button in a GUI

**To Reproduce**
1) Make a GUI with skquery
2) Open the GUI
3) Click on a button
4) Admire the error in your console

**Information:**
- Skript version: 2.4-alpha1, 2.4-alpha2, 2.4-alpha3
- Server software: PaperSpigot
- Server version: 1.14.2

**Error:**
` [18:36:41 ERROR]: Could not pass event InventoryClickEvent to Skript v2.4-alpha1
java.lang.NoSuchMethodError: org.bukkit.inventory.Inventory.getTitle()Ljava/lang/String;
at com.w00tmast3r.skquery.elements.expressions.ExprInventoryName.convert(ExprInventoryName.java:32) ~[?:?]
at com.w00tmast3r.skquery.elements.expressions.ExprInventoryName.convert(ExprInventoryName.java:19) ~[?:?]
at ch.njol.skript.registrations.Converters.convert(Converters.java:343) ~[?:?]
at ch.njol.skript.registrations.Converters.convertUnsafe(Converters.java:334) ~[?:?]
at ch.njol.skript.expressions.base.PropertyExpression.get(PropertyExpression.java:104) ~[?:?]
at ch.njol.skript.expressions.base.SimplePropertyExpression.get(SimplePropertyExpression.java:55) ~[?:?]
at ch.njol.skript.expressions.base.PropertyExpression.get(PropertyExpression.java:75) ~[?:?]
at ch.njol.skript.lang.util.SimpleExpression.check(SimpleExpression.java:161) ~[?:?]
at ch.njol.skript.conditions.CondCompare.check(CondCompare.java:253) ~[?:?]
at ch.njol.skript.lang.Condition.run(Condition.java:56) ~[?:?]
at ch.njol.skript.lang.Conditional.walk(Conditional.java:51) ~[?:?]
at ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:89) ~[?:?]
at ch.njol.skript.lang.Trigger.execute(Trigger.java:57) ~[?:?]
at ch.njol.skript.SkriptEventHandler.check(SkriptEventHandler.java:156) ~[?:?]
at ch.njol.skript.SkriptEventHandler$1.execute(SkriptEventHandler.java:114) ~[?:?]
at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.14.2.jar:git-Paper-59]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.14.2.jar:git-Paper-59]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:534) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:2340) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:33) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:18) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:127) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:105) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.MinecraftServer.aW(MinecraftServer.java:998) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:991) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:115) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.MinecraftServer.sleepForTick(MinecraftServer.java:975) ~[patched_1.14.2.jar:git-Paper-59]
at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:908) ~[patched_1.14.2.jar:git-Paper-59]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]`

**Comment by TheLimeGlass** [comment]
getTitle doesn't exist in 1.14. Skript 3.4-alpha1 just completely removed it.

Three things that can happen:
1.) I can add support to read the inventory of the event and make this syntax work ONLY if the event is within the inventory click event and the inventory being checked is the same from the event.
2.) Skript or SkQuery add a new Inventory View event value and new syntax for it.
3.) *User fixable* rather than checking the name of the inventory to be sure it is the same inventory, you can add the user to a list variable, and check if the user is within the list inventory.
```
open inventory#etc
add player to {temp::customGUI::open::*}

on inventory click:
    event-inventory is not player's inventory
    {temp::customGUI::open::*} contains player
    cancel event
    #do your handling for this inventory

on inventory close:
    remove player from {temp::customGUI::open::*}

on script unload:
    delete {temp::*}
```

**Comment by Laukage** [comment]
> getTitle doesn't exist in 1.14. Skript 2.4-alpha1 just completely removed it until someone fixes it.

Why was it removed? and can't you just suggest it being added back in?

**Comment by TheLimeGlass** [comment]
No it's Minecraft, you can't just tell them to fix something, as Bukkit isn't run by Mojang anymore.

**Comment by YellowZaki** [comment]
@TheLimeGlass Mmm, Inventory.getTitle() (deprecated since 1.13) was replaced by InventoryView.getTitle()
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/inventory/InventoryView.html#getTitle--

**Comment by Laukage** [comment]
Any ETA on this?

**Comment by TheLimeGlass** [comment]
Fixed in version 4.1.0.
https://github.com/SkQuery/SkQuery/commit/4920c1b98ea628fafb1fa5e1ba4608adb2829fc7
https://www.spigotmc.org/resources/skquery.36631/update?update=287266
